contracts/token/ERC721/ERC721.sol

```diff
@@ -221,40 +199,73 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
     }
 
     /**
-     * @dev Returns whether `spender` is allowed to manage `tokenId`.
-     *
-     * Requirements:
-     *
-     * - `tokenId` must exist.
+     * @dev Returns the approved address for `tokenId`. Returns 0 if `tokenId` is not minted.
      */
-    function _isApprovedOrOwner(address spender, uint256 tokenId) internal view virtual returns (bool) {
-        address owner = ownerOf(tokenId);
-        return (spender == owner || isApprovedForAll(owner, spender) || getApproved(tokenId) == spender);
+    function _getApproved(uint256 tokenId) internal view virtual returns (address) {
+        return _tokenApprovals[tokenId];
     }
 
     /**
-     * @dev Safely mints `tokenId` and transfers it to `to`.
+     * @dev Returns whether `spender` is allowed to manage `owner`'s tokens, or `tokenId` in
+     * particular (ignoring whether it is owned by `owner`).
      *
-     * Requirements:
-     *
-     * - `tokenId` must not exist.
-     * - If `to` refers to a smart contract, it must implement {IERC721Receiver-onERC721Received}, which is called upon a safe transfer.
+     * WARNING:
+     * - ownership is not checked.
+     * - spender = address(0) will lead to this function returning true in unexpected situations.
+     */
+    function _isApproved(address owner, address spender, uint256 tokenId) internal view virtual returns (bool) {
```

Comment:
I'm concerned that the meaning of `_isApproved`/`_checkApproved` is very different from the meaning of `_getApproved`.


I would rename `_is/checkApproved` to `_is/checkOperator`.

Reply:
There's already a `_checkApproved` so it will conflict.

What about the convention of?:

1. `_is*` for boolean returning values (eg. `_isApproved`)
2. `_check`for functions reverting when `_is*` functions return false (eg. `_checkApproved`)

Reply:
Yes this is the convention we seem to be using.